src/hera/workflows/task.py

```diff
@@ -71,22 +123,59 @@ def _get_dependency_tasks(self) -> List[str]:
     def _subtype(self) -> str:
         return "tasks"
 
-    def next(self, other: Task, operator: Operator = Operator.and_, on: Optional[TaskResult] = None) -> Task:
+    def next(
+        self,
+        other: Task,
+        operator: Optional[Operator] = None,
+        on: OnType = None,
```

Comment:
Note the change of types is backwards compatible and the behaviour is unchanged (the default `Operator.and_` is now stored as a class variable).